lib/fastly_nsq/message_queue/listener.rb

```diff
@@ -26,14 +26,14 @@ def process_next_message
 
     private
 
+    attr_reader :channel, :topic
+
     def process_one_message
       message = consumer.pop
-      MessageProcessor.new(message.body).go
+      MessageProcessor.new(topic: topic,  message_body: message.body).go
```

Comment:
OMG how embarrassing. Fixed here and elsewhere in 8e65433, only for aesthetic reasons.